processing/src/main/java/io/druid/query/lookup/LookupBean.java

```diff
@@ -19,32 +19,76 @@
 
 package io.druid.query.lookup;
 
+import com.fasterxml.jackson.annotation.JsonCreator;
 import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.base.Preconditions;
 
 //TODO merge this code to the same definition when pr/1576 is merged
 class LookupBean
 {
+  private final LookupExtractorFactoryContainer container;
+  private final String name;
+
+  @JsonCreator
+  public LookupBean(
+      @JsonProperty("name") String name,
+      //kept for backward compatibility with druid ver <= 0.9.2 persisted snapshots
+      @Deprecated @JsonProperty("factory") LookupExtractorFactory factory,
+      @JsonProperty("container") LookupExtractorFactoryContainer container
+  )
+  {
+    Preconditions.checkArgument(factory == null || container == null, "only one of factory or container should exist");
```

Comment:
well there is another line after this that checks the other case.
`Preconditions.checkArgument(factory != null || container != null, "one of factory or container must exist");`

but will replace those 2 lines with XOR , that is definitely more concise